### PR TITLE
Change GBM testing from referring by index to referring by feature name.

### PR DESCRIPTION
**Pull request by criver**
@bayesian

### CODE REVIEW

**Comment by criver** [comment]
Tested on a data set. The post-PR eval output matches pre-PR one.

File: Tree.h [hunk 1]
```diff
@@ -132,13 +132,21 @@ class LeafNode : public TreeNode<T> {
...
 // load a regression tree from Json
 template <class T>
-TreeNode<T>* fromJson(const folly::dynamic& obj) {
-  int index = obj["index"].asInt();
+TreeNode<T>* fromJson(const folly::dynamic& obj, const Config& cfg) {
+  const folly::dynamic* feature = nullptr;
+  try {
+    feature = &obj["feature"];
+  } catch(...) {
+  }
```
Comment (bayesian) [inline Tree.h]: check folly obj has key "feature" or "index", thus no need for try and empty catch.. 
and also keep the backward compatibility.

Comment (criver) [inline Tree.h]: Are you suggesting use **find**? I tried **find** but got the following compilation error:

/usr/local/include/folly/dynamic-inl.h:664:32: error: ‘get’ is not a member of ‘folly::dynamic::GetAddrImpl<folly::dynamic::const_item_iterator>’
   return GetAddrImpl<T>::get(u_);
                                ^

Comment (bayesian) [inline Tree.h]: it should work, https://github.com/facebook/folly/blob/master/folly/docs/Dynamic.md

auto pos = obj.find("no_such_key);
    // pos == obj.items().end()

the folly is local built, we built in chef cookbook:

bash 'install_folly' do
  cwd '/mnt/projects/folly/folly/'
  user "root"
  code <<-CMDS                                                                                                                          
     autoreconf -ivf                                                                                                                    
     ./configure --with-boost=/usr/local CPPFLAGS="-I/usr/local/include" LDFLAGS="-L/usr/local/lib" GFLAGS_NAMESPACE=google             
     make && make check && make install                                                                                                 
  CMDS                                                                                                                                  
end

**Comment by bayesian** [comment]
LGTM.. I could fix that later.

File: Tree.h (hunk 1 above)
Comment (criver) [inline Tree.h]: I think there might be a bug in dynamic. The error message is that get is not implemented for const_iterator. The example works but in our code we pass in a const reference. Will submit it for now.

**Pull request merged**